### PR TITLE
Main deck: comment and trick (video) pass

### PR DESCRIPTION
**Pull request by Miepee**
- Adds TODO's and FIXME's to things that are worth remembering to change/revise later
- Adds more comments
- Adds a lot of trick videos
- Adds events for the operations deck missile shield, and maintenance shaft gerons
- Adds a lot of shinespark tricks

As well as a lot of other Main Deck clean up.
Note, that for open transitions, or hatches, I removed `Disabled Door Lock Rando`, because no doors can be placed there. In a future PR, I plan to add "shuffle open hatches" misc resource, so that shinesparks aren't always completely out of logic when Door Lock Rando is enabled.

### CODE REVIEW

File: randovania/games/fusion/logic_database/header.txt [hunk 1]
```diff
@@ -41,6 +41,7 @@ Templates
 * Can Kill Missile Geron:
       Any of the following:
           Any of the following:
+              # TODO: missile gerons actually take only 3 missiles? is this for the s1 cube gerons instead?
```
Comment (JeffGainsNGames) [inline randovania/games/fusion/logic_database/header.txt:44]: gerons have 30HP
image

File: randovania/games/fusion/logic_database/Main Deck.txt [hunk 2]
```diff
@@ -334,11 +350,11 @@ Extra - room_id: [13, 74, 85]
   > Event - SA-X
       All of the following:
           All of the following:
-              # SA-X Requirements
+              # SA-X Requirements - TODO: revise these. iirc plasma isnt needed anymore.
               Charge Beam and Plasma Beam
               Wide Beam or Combat (Advanced)
           Any of the following:
-              # Core-X Requirements
+              # Core-X Requirements - TODO: how many missiles?
```
[the remark this answers is not in the document]
Comment (Miepee) [reply randovania/games/fusion/logic_database/Main Deck.txt:357]: Trickless should not need to farm missiles at sa-x corex

File: randovania/games/fusion/logic_database/Main Deck.txt [hunk 3]
```diff
@@ -351,10 +367,16 @@ Extra - room_id: [13, 74, 85]
               Combat (Advanced) and Normal Damage ≥ 400
               Combat (Expert) and Normal Damage ≥ 150
           All of the following:
-              # Dodging Requirements
+              # Dodging Requirements - TODO: how does screw attack help?
```
[the remark this answers is not in the document]
Comment (Miepee) [reply randovania/games/fusion/logic_database/Main Deck.txt:370]: ah right. will remove

File: randovania/games/fusion/logic_database/Main Deck.txt [hunk 4]
```diff
@@ -1522,6 +1633,7 @@ Extra - room_id: [58]
 
 > Save Station; Heals? False; Spawn Point
   * Layers: default
+  * TODO: does this require the reactor to be activated in order to get used?
```
[the remark this answers is not in the document]
Comment (Miepee) [reply randovania/games/fusion/logic_database/Main Deck.txt:1636]: right, i remember baria being confused at it in a seed. will remove

File: randovania/games/fusion/logic_database/header.txt (hunk 1 above)
Comment (Miepee) [reply randovania/games/fusion/logic_database/header.txt:44]: i forgot that this is for theoretical shoot missiles without main missile data. will remove

File: randovania/games/fusion/logic_database/Main Deck.txt [hunk 5]
```diff
@@ -666,7 +697,8 @@ Extra - room_id: [23]
   > Door to Main Elevator Access
       Any of the following:
           Morph Ball
-          Speed Booster and Shinespark Tricks (Intermediate) and Disabled Door Lock Rando and Disabled Entrance Rando
+          # Shinespark from the Navigation Station: https://www.youtube.com/watch?v=QcD2-t9UDrg
+          Speed Booster and Shinespark Tricks (Advanced) and Disabled Entrance Rando
```
[the remark this answers is not in the document]
Comment (Miepee) [reply randovania/games/fusion/logic_database/Main Deck.txt:701]: I didn't see that s1 shinespark was intermediate when i made this, i'll bump this back down.

File: randovania/games/fusion/logic_database/Main Deck.txt [hunk 6]
```diff
@@ -959,7 +992,12 @@ Extra - room_id: [35]
   * Open Passage to Operations Deck/Other to Operations Maintenance Shaft
   * Extra - door_idx: (79,)
   > Other to Crew Quarters East
-      Can Kill Missile Geron
+      After Main Deck Operations Deck Maintenance Shaft Upper Missile Geron
+  > Event - Upper Maintainance Geron
+      Any of the following:
+          Can Kill Missile Geron
+          # Shinespark from Operations Deck: https://www.youtube.com/watch?v=Y3k5lGKTX4Y
+          Level 0 Locks and Speed Booster and After Main Deck Operations Deck Missile Shield and Shinespark Tricks (Intermediate) and Disabled Door Lock Rando and Disabled Entrance Rando
```
Comment (Miepee) [inline randovania/games/fusion/logic_database/Main Deck.txt:1000]: this needs beginner knowledge

File: randovania/games/fusion/logic_database/Main Deck.txt [hunk 7]
```diff
@@ -973,15 +1011,34 @@ Extra - room_id: [35]
   * L0 Hatch to Crew Quarters West/Door to Operations Maintenance Shaft
   * Extra - door_idx: (81,)
   > Door to Crew Quarters East
-      Can Kill Missile Geron
+      After Main Deck Operations Deck Maintenance Shaft Lower Missile Geron
+  > Event - Lower Maintainance Geron
+      Any of the following:
+          Can Kill Missile Geron
+          # Shinespark from Main Deck Entrance: https://www.youtube.com/watch?v=dWJXG-nGo6A
+          Level 0 Locks and Speed Booster and Shinespark Tricks (Hypermode) and Wall Jump (Advanced) and Disabled Door Lock Rando and Disabled Entrance Rando
```
Comment (Miepee) [inline randovania/games/fusion/logic_database/Main Deck.txt:1019]: this needs beginner knowledge

**Review by JeffGainsNGames — APPROVED** [review]

**Pull request merged**